buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -0,0 +1,1428 @@
+            int startIndex = readerIndex();
+            skipBytes(size);
+            buf.init(delegate, this, 0, 0, startIndex, size, maxCapacity);
+            return buf;
+        }
+
+        @Override
+        protected byte _getByte(int index) {
+            return delegate.getByte(index);
+        }
+
+        @Override
+        protected short _getShort(int index) {
+            return delegate.getShort(index);
+        }
+
+        @Override
+        protected short _getShortLE(int index) {
+            return delegate.getShortLE(index);
+        }
+
+        @Override
+        protected int _getUnsignedMedium(int index) {
+            return delegate.getUnsignedMedium(index);
+        }
+
+        @Override
+        protected int _getUnsignedMediumLE(int index) {
+            return delegate.getUnsignedMediumLE(index);
+        }
+
+        @Override
+        protected int _getInt(int index) {
+            return delegate.getInt(index);
+        }
+
+        @Override
+        protected int _getIntLE(int index) {
+            return delegate.getIntLE(index);
+        }
+
+        @Override
+        protected long _getLong(int index) {
+            return delegate.getLong(index);
+        }
+
+        @Override
+        protected long _getLongLE(int index) {
+            return delegate.getLongLE(index);
+        }
+
+        @Override
+        protected void _setByte(int index, int value) {
+            delegate.setByte(index, value);
+        }
+
+        @Override
+        protected void _setShort(int index, int value) {
+            delegate.setShort(index, value);
+        }
+
+        @Override
+        protected void _setShortLE(int index, int value) {
+            delegate.setShortLE(index, value);
+        }
+
+        @Override
+        protected void _setMedium(int index, int value) {
+            delegate.setMedium(index, value);
+        }
+
+        @Override
+        protected void _setMediumLE(int index, int value) {
+            delegate.setMediumLE(index, value);
+        }
+
+        @Override
+        protected void _setInt(int index, int value) {
+            delegate.setInt(index, value);
+        }
+
+        @Override
+        protected void _setIntLE(int index, int value) {
+            delegate.setIntLE(index, value);
+        }
+
+        @Override
+        protected void _setLong(int index, long value) {
+            delegate.setLong(index, value);
+        }
+
+        @Override
+        protected void _setLongLE(int index, long value) {
+            delegate.setLongLE(index, value);
+        }
+
+        @Override
+        public int capacity() {
+            return delegate.capacity();
+        }
+
+        @Override
+        public ByteBuf capacity(int newCapacity) {
+            delegate.capacity(newCapacity);
+            return this;
+        }
+
+        @Override
+        public ByteBufAllocator alloc() {
+            return delegate.alloc();
```

Comment:
is this correct and ensure we always return the `AdaptiveByteBufAllocator` instance ?

Reply:
Yes, this is given by the AdaptiveByteBufAllocator ChunkAllocator implementations.